sklearn/linear_model/_sgd_fast.pyx

```diff
@@ -654,7 +656,7 @@ def _plain_sgd(cnp.ndarray[double, ndim=1, mode='c'] weights,
                     best_score = score
             # or evaluate the loss on the training set
             else:
-                if tol > -INFINITY and sumloss > best_loss - tol * n_samples:
+                if tol > -INFINITY and sumloss > best_loss - tol * train_count:
```

Comment:
Since this changes the stopping condition compared to `main`, I think this requires a "Changed models" entry here:

https://github.com/scikit-learn/scikit-learn/blob/213d21fe719ce5778726203893c78251b8af34fa/doc/whats_new/v1.2.rst?plain=1#L14-L15